packages/rest/src/validation/request-body.validator.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  RequestBodyObject,
+  SchemaObject,
+  SchemasObject,
+  isReferenceObject,
+} from '@loopback/openapi-v3-types';
+import * as AJV from 'ajv';
+import * as debugModule from 'debug';
+import * as util from 'util';
+import {HttpErrors} from '..';
+import {RestHttpErrors} from '../coercion/rest-http-error';
+
+const toJsonSchema = require('openapi-schema-to-json-schema');
+
+const debug = debugModule('loopback:rest:validation');
+
+export function validateRequestBody(
+  // tslint:disable-next-line:no-any
+  body: any,
+  requestBodySpec: RequestBodyObject | undefined,
+  globalSchemas?: SchemasObject,
+) {
+  if (requestBodySpec && requestBodySpec.required && body == undefined)
+    throw new HttpErrors.BadRequest('Request body is required');
+
+  const schema = getRequestBodySchema(requestBodySpec, globalSchemas || {});
+  debug('Request body schema: %j', util.inspect(schema, {depth: null}));
+  if (!schema) return;
+
+  const jsonSchema = convertToJsonSchema(schema);
+  validateValueAgainstJsonSchema(body, jsonSchema);
+}
+
+function getRequestBodySchema(
+  requestBodySpec: RequestBodyObject | undefined,
+  globalSchemas: SchemasObject,
+): SchemaObject | undefined {
+  if (!requestBodySpec) return;
+
+  const content = requestBodySpec.content;
+  // FIXME(bajtos) we need to find the entry matching the content-type
```

Comment:
👍 Good catch, doesn't look a big code change, opened an issue to track it: https://github.com/strongloop/loopback-next/issues/1494
Will try to address it in the next PR(in a reasonable time)